server/models/admin.model.js

```diff
@@ -45,4 +45,35 @@ const admin = new Schema({
     }
 }, { timestamps: true });
 
+// Password Hash Middleware
+admin.pre('save', function(next) {
+  const admin = this;
+  if (!admin.isModified('password')) {
+    return next()
+  }
+  bcrypt.hash(admin.password, 10, function(err, hash) {
+      if (err) return next(err);
+      admin.password = hash;
+      next();
+  });
+});
+
+// Helper for comparing password
+admin.methods.comparePassword = function comparePassword(adminPassword,cb) {
+    bcrypt.compare(adminPassword, this.password, (err, isMatch) => {
+       cb(err, isMatch)
+    })
+}
+
+admin.options.toJSON = {
+    transform (doc, ret, options) {
+      delete ret._id
```

Comment:
I feel it would be nice if you could place this code in folder `middlwares/admin.js` instead of inside admin model.
let me know if that will causes any issues..